### PR TITLE
Input time with tz offset pt 2

### PR DESCRIPTION
**Pull request by pixelbandito**
- fixes an issue with min/max validation in the `AsDate` component.
- refactor: changes the interface for a few of the `InputTime/utils`, so there's a slight refactor there.
- Adds named export for `AsDate`: `InputTimeAsDate`
- Adds named export for `AsString`: `InputTimeAsString`
- docs: Most of the huge diff comes from updating the `InputTime` story to add sections for the `AsDate`/`AsString` components.

### CODE REVIEW

File: src/components/InputTime/utils.test.tsx [hunk 1]
```diff
@@ -381,7 +382,8 @@ describe('getDateTimeFromShortTimeString', () => {
         timeZoneName: 'America/Chicago',
       });
 
-      const dateIST = getDateTimeFromShortTimeString('12:30', {
+      const dateIST = getDateTimeFromShortTimeString({
+        time: '12:30',
         timeZoneOffset: +330,
       });
```
Comment (dkordik) [inline src/components/InputTime/utils.test.tsx:389]: some good edge cases to add tests for:
- offset brings you into the previous day
- offset brings you into the next day
- offset brings you into the next day during daylight savings time? :shudder:

Comment (pixelbandito) [reply src/components/InputTime/utils.test.tsx:389]: Curses. That is a very good idea.

**Comment by pixelbandito** [comment]
Hard nope until some underlying kinks are worked out.
